ServantPersistent/src/DBApi.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE DataKinds       #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeOperators   #-}
+{-# LANGUAGE TypeApplications   #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
+{-# LANGUAGE TypeInType #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DeriveAnyClass #-}
+module DBApi
+    where
+
+import Data.Aeson
+import Data.Aeson.TH
+import Database.Persist.Sql
+import Database.Persist.TH
+import GHC.Generics
+import Data.Time.Clock
+import Data.Time
+import Data.Text
+import Types
+import Control.Monad.IO.Class
+import Control.Monad.Reader
+import Control.Lens
+import Data.ByteString
+import Data.Proxy
+import Models
+import Safe
+
+type TenantOutput = TenantBase TenantOutputField
+
+data TenantOutputField = TOF { status :: TenantStatus
+                             , ownerId :: Maybe DBUserId
+                             } deriving (Generic, ToJSON)
+
+tbdbIso :: Iso' (TenantBase TenantOutputField) DBTenant
+tbdbIso = iso (\(TB ti c u (TOF status id)) -> DBTenant (ti ^. name) (ti ^. backofficeDomain) id status c u) (\dbt -> TB (dbt ^. tenantIdent) (dbt ^. createdAt) (dbt ^. updatedAt) (TOF (dbt ^. dBTenantStatus)  (dbt ^. dBTenantOwnerId)))
+
+dbGetTenant :: TenantID -> App (Maybe TenantOutput)
```

Comment:
This needs a lot of explanation. Why the need to construct `TenantBase` and all the related types? What other alternatives were considered? Pros and cons of each?

Why can the DB API not return a DBTenant?